### PR TITLE
Cosmic CMS updates

### PR DESCRIPTION
**Pull request by tonyspiro**
## Updates

- Update packages
- Convert to TypeScript

### CODE REVIEW

File: examples/cms-cosmic/components/alert.tsx [hunk 1]
```diff
@@ -40,3 +45,5 @@ export default function Alert({ preview }) {
     </div>
   )
 }
+
+export default Alert
```
Comment (balazsorban44) [inline examples/cms-cosmic/components/alert.tsx:49]:
```suggestion
```

File: examples/cms-cosmic/pages/api/exit-preview.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+const exit = async (_, res) => {
   // Exit the current user from "Preview Mode". This function accepts no args.
   res.clearPreviewData()
 
   // Redirect the user back to the index page.
   res.writeHead(307, { Location: '/' })
   res.end()
 }
+
+export default exit
```
Comment (balazsorban44) [inline examples/cms-cosmic/pages/api/exit-preview.ts:10]: this seems to be an unnecessary change

File: examples/cms-cosmic/pages/api/preview.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { getPreviewPostBySlug } from '@/lib/api'
 
-export default async function preview(req, res) {
+const preview = async (req, res) => {
```
Comment (balazsorban44) [inline examples/cms-cosmic/pages/api/preview.ts:3]: this seems to be an unnecessary change, let's revert

File: examples/cms-cosmic/pages/posts/[slug].tsx [hunk 4]
```diff
@@ -52,24 +61,34 @@ export default function Post({ post, morePosts, preview }) {
...
-      morePosts: data.morePosts || [],
-    },
+    }
+  } catch (err) {
+    return <ErrorPage statusCode={err.status} />
   }
 }
 
-export async function getStaticPaths() {
+export const getStaticPaths = async () => {
```
Comment (balazsorban44) [inline examples/cms-cosmic/pages/posts/[slug].tsx:91]: this seems to be an unnecessary change, let's revert

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks. Generally, I think we would like to keep the `function` declarations. See https://github.com/vercel/next.js/pull/38091#discussion_r911320691

So let's revert those changes. :+1:

**Comment by tonyspiro** [comment]
> Thanks. Generally, I think we would like to keep the `function` declarations. See [#38091 (comment)](https://github.com/vercel/next.js/pull/38091#discussion_r911320691)
> 
> So let's revert those changes. 👍

@balazsorban44 this should be ready for re review, thank you!

**Comment by tonyspiro** [comment]
Any update @balazsorban44?

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
Thanks!

**Comment by tonyspiro** [comment]
Any update on this @balazsorban44?

**Comment by balazsorban44** [comment]
Hi @tonyspiro, it seems like your branch does not allow our bot (or me) to merge this PR. Can you make sure you allow contributors to edit so we can merge it?

**Comment by tonyspiro** [comment]
> Hi @tonyspiro, it seems like your branch does not allow our bot (or me) to merge this PR. Can you make sure you allow contributors to edit so we can merge it?

I just added you to the repo. Please let me know if you need anything else.

**Comment by balazsorban44** [comment]
Update @tonyspiro

Turns out it's because of this: https://github.com/orgs/community/discussions/5634

The GitHub UI says there are merge conflicts, but I am not able to see them. Can you fix those? :pray:

image

**Pull request closed without merging**